spec/sec-properties-of-the-regexp-prototype-object-patch.html

```diff
@@ -71,27 +72,28 @@ <h1>Runtime Semantics: RegExpBuiltinExec ( _R_, _S_ )</h1>
             1. <del>Let _capturedValue_ be the String value consisting of the code units of _captureI_.</del>
           1. <ins>If _captureI_ is *undefined*, then</ins>
             1. <ins>Let _capturedValue_ be *undefined*.</ins>
-            1. <ins>Add *undefined* as the last element of _indices_.</ins>
+            1. <ins>If _indices_ is a List, append *undefined* to _indices_.</ins>
           1. <ins>Else,</ins>
             1. <ins>Let _captureStart_ be _captureI_'s _startIndex_.</ins>
             1. <ins>Let _captureEnd_ be _captureI_'s _endIndex_.</ins>
             1. <ins>If _fullUnicode_ is *true*, then</ins>
               1. <ins>Set _captureStart_ to ! GetStringIndex(_S_, _Input_, _captureStart_).</ins>
               1. <ins>Set _captureEnd_ to ! GetStringIndex(_S_, _Input_, _captureEnd_).</ins>
             1. <ins>Let _capture_ be the Match { [[StartIndex]]: _captureStart_, [[EndIndex]:: _captureEnd_ }.</ins>
-            1. <ins>Append _capture_ to _indices_.</ins>
             1. <ins>Let _capturedValue_ be ! GetMatchString(_S_, _capture_).</ins>
+            1. <ins>If _indices_ is a List, append _capture_ to _indices_.</ins>
           1. Perform ! CreateDataProperty(_A_, ! ToString(_i_), _capturedValue_).
           1. If the _i_th capture of _R_ was defined with a |GroupName|, then
             1. Let _s_ be the StringValue of the corresponding |RegExpIdentifierName|.
             1. Perform ! CreateDataProperty(_groups_, _s_, _capturedValue_).
-            1. <ins>Assert: _groupNames_ is a List.</ins>
-            1. <ins>Append _s_ to _groupNames_.</ins>
+            1. <ins>If _indices_ is a List, then</ins>
+              1. <ins>Assert: _groupNames_ is a List.</ins>
+              1. <ins>Append _s_ to _groupNames_.</ins>
           1. <ins>Else,</ins>
             1. <ins>If _groupNames_ is a List, append *undefined* to _groupNames_.</ins>
-        1. <ins>Let _indicesArray_ be MakeIndicesArray(_S_, _indices_, _groupNames_).</ins>
-        1. <ins>Perform ! CreateDataProperty(_A_, `"indices"`, _indicesArray_).</ins>
-        1. <ins>DRAFT NOTE: While the `"indices"` data property is created eagerly in the spec, an implementation may choose to lazily create it if the laziness is unobservable.</ins>
+        1. <ins>If _indices_ is a List, then</ins>
+          1. <ins>Let _indicesArray_ be MakeIndicesArray(_S_, _indices_, _groupNames_).</ins>
```

Comment:
```suggestion
          1. <ins>Let _indicesArray_ be ! MakeIndicesArray(_S_, _indices_, _groupNames_).</ins>
```
or ?, as appropriate

Reply:
Every invocation of an operation in `MakeIndicesArray` uses `!`, so this should probably use `!` too.